### PR TITLE
[magiclysm] Goblin encampment

### PR DESCRIPTION
**Pull request by KorGgenT**
#### Summary

```SUMMARY: Mods "[magiclysm] Goblin encampment"```
#### Purpose of change
Adds a goblin encampment and goblin monsters. Part of a general idea to add more mid game locations for magiclysm.

#### Describe the solution
pretty straightforward: add goblin slingers, warriors, and chieftains, and make an encampment area for them to spawn in with some loot.

#### Testing
spawned in the special to make sure it didn't error out, and looked ok.

#### Additional context
Could not have done this without @curstwist ! Thanks a bunch!

image

### CODE REVIEW

File: data/mods/Magiclysm/monsters/goblin.json [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+    "speed": 105,
+    "material": [ "flesh" ],
+    "symbol": "g",
+    "color": "brown",
+    "aggression": 100,
+    "morale": 100,
+    "melee_skill": 4,
+    "melee_dice": 2,
+    "melee_dice_sides": 4,
+    "melee_bash": 2,
```
Comment (ZhilkinSerg) [inline data/mods/Magiclysm/monsters/goblin.json:22]:
```suggestion
```

File: data/mods/Magiclysm/worldgen/goblin_encampment.json [hunk 2]
```diff
@@ -0,0 +1,578 @@
...
+        "         .....................................           ....................................................           ",
+        "         .....................................           ....................................................           ",
+        "         .....................................           ....................................................           ",
+        "         .......>                                                    >                                .......           ",
+        "         .......                                                                                      .......           ",
+        "         .......                                                                                      .......           ",
+        "         .......                                                                                      .......           ",
+        "         .......                                                                                      .......           ",
+        "         .......         *                       *                       *                       *    .......           ",
```
Comment (ZhilkinSerg) [inline data/mods/Magiclysm/worldgen/goblin_encampment.json:517]: Looks like `*` is not defined in mapgen palette.

Comment (curstwist) [reply data/mods/Magiclysm/worldgen/goblin_encampment.json:517]: these were map boundaries I forgot to remove so they can be deleted

File: data/mods/Magiclysm/worldgen/goblin_encampment.json [hunk 3]
```diff
@@ -0,0 +1,578 @@
...
+    "furniture": {
+      "h": "f_straw_bed",
+      "@": "f_logstool",
+      "c": "f_firering",
+      "Y": [ [ "f_boulder_small", 5 ], "f_boulder_medium", "f_boulder_large" ]
+    },
+    "items": {
+      "c": { "item": "goblin_cookpot", "chance": 50 },
+      "h": { "item": "bed", "chance": 40, "repeat": [ 1, 2 ] },
+      "!": { "item": "lair_loot_generic", "chance": 40, "repeat": [ 1, 2 ] },
```
Comment (curstwist) [inline data/mods/Magiclysm/worldgen/goblin_encampment.json:137]: maybe swap to `lair_black_dragon`, this is still pretty low tier loot but with the possibility of some weapons and rare chances of magiclysm themed loot.

**Comment by kwl01skz** [comment]
awesome

**Pull request merged**